lessons/unittest/original/module.py

```diff
@@ -0,0 +1,35 @@
+''' 
+  A class we can use to test unittesting
+'''
+
+import numpy as np
+
+
+class Var(object):
+  ''' A container class for numpy arrays and some meta data '''
+
+  def __init__(self, name, data, axes=None, units=None):
+    ''' construct variable object with meta data '''
+    self.name = name
+    self.units = units
+    self.axes = axes
+    self.data = data
+    
+  def __iadd__(self, data):
+    ''' overload in-place addition function: add number or arrays '''
+    self.data += data
+    return self
+  
+  def __add__(self, other):
+    ''' overload addition function: add arrays and create new variable '''
+    # check input
+    if not isinstance(other, Var): raise Exception
```

Comment:
Interesting; I didn't know that this NotImplemented actually raises a sensible exception. In my own code I have different error classes (just different names); that's why I used an Exception; I usually use TypeError for more generic type mismatches.